framework/src/org/checkerframework/common/value/ValueVisitor.java

```diff
@@ -26,20 +30,49 @@
  */
 public class ValueVisitor extends BaseTypeVisitor<ValueAnnotatedTypeFactory> {
 
+    /** helper set that includes all integer literal kinds */
+    private final Set<Kind> intLiteralKinds =
+            EnumSet.of(Kind.INT_LITERAL, Kind.LONG_LITERAL, Kind.CHAR_LITERAL);
+
     public ValueVisitor(BaseTypeChecker checker) {
         super(checker);
     }
 
+    /** helper function to determine if a given kind is integer literal */
+    private boolean isIntLiteral(Kind k) {
+        return intLiteralKinds.contains(k);
+    }
+
+    /** get value from a give expression tree, assuming the tree kind is int literal */
+    private long getIntLiteralValue(ExpressionTree exp) {
+        switch (exp.getKind()) {
+            case INT_LITERAL:
+                return ((Number) ((LiteralTree) exp).getValue()).longValue();
+            case LONG_LITERAL:
+                return (long) ((LiteralTree) exp).getValue();
+            case CHAR_LITERAL:
+                return (long) ((Character) ((LiteralTree) exp).getValue());
+            default:
+                throw new IllegalArgumentException(
+                        "exp should be within the covered kinds (INT_LITERAL, LONG_LITERAL, CHAR_LITERAL");
+        }
+    }
+
     @Override
     protected ValueAnnotatedTypeFactory createTypeFactory() {
         return new ValueAnnotatedTypeFactory(checker);
     }
 
     /**
-     * Issues a warning if any constant-value annotation has &gt; MAX_VALUES number of values
+     * Issues an error if any @IntRange annotation has its 'from' value greater than 'to' value.
+     * Works together with
+     * ValueAnnotatedTypeFactory.ValueTypeAnnotator.replaceWithUnknownValInSpecialCases which treats
```

Comment:
When referring to another method, please use the `{@link ...}` Javadoc markup.
(Like many comments, this one applies throughout the code.)